client/app/bundles/course/reference-timelines/components/DayCalendar/DayColumn.tsx

```diff
@@ -0,0 +1,43 @@
+import { CSSProperties, memo } from 'react';
+import { areEqual } from 'react-window';
+import { Typography } from '@mui/material';
+import moment from 'moment';
+
+import { getSecondsFromDays, isToday, isWeekend } from '../../utils';
+
+interface DayProps {
+  index: number;
+  style: CSSProperties;
+}
+
+const DayColumn = (props: DayProps): JSX.Element => {
+  const day = moment.unix(getSecondsFromDays(props.index));
+
+  return (
+    <div
+      key={day.toString()}
+      className={`select-none rounded-t-lg ${
+        isWeekend(day) ? 'bg-neutral-50' : ''
+      } ${isToday(day) ? 'bg-red-50' : ''}`}
+      style={props.style}
+    >
+      <div
+        className={`sticky top-0 z-20 flex h-20 shrink-0 flex-col justify-end bg-slot-1 slot-1-white ${
```

Comment:
I think initially the reason for disabling `flex-shrink` is because when the number of days appeared in the container is almost equal to the width, this div tends to stretch and fill the space.

I set the z-index here to ensure if someone puts a new element around it, the day will always be on top (e.g., a floating month indicator above it; did not manage to ship this).